Form/FormMapper.php

```diff
@@ -275,6 +280,21 @@ public function addHelp($name, $help)
     }
 
     /**
+     * Symfony default form class sadly can't handle
+     * form element with dots in its name (when data
+     * get bound, the default dataMapper is a PropertyPathMapper).
+     * So use this trick to avoid any issue.
+     *
+     * @param string $fieldName
+     *
+     * @return string
+     */
+    public function sanitizeFieldName($fieldName)
+    {
+        return str_replace(array('__', '.'), array('____', '__'), $fieldName);
```

Comment:
Whatever character we choose to convert dot we need to escape this one to avoid conflict.
If we convert `.` to `____` there will be conflict in form like this one while `____` isn't escaped.
```
$formMapper->add('fo.o')->add('fo____o')
```